### PR TITLE
docs: refactored the contributing guide

### PR DESCRIPTION
**Pull request by anikethsaha**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [x] This is a **docs update**
- [x] This is a **metadata update**

### For Bugs and Features; did you add new tests?
NA

### Motivation / Use-Case
Changed the npm script and added commit guide

### Breaking Changes
No

### Additional Info
NA

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -52,10 +52,55 @@ Pull requests often need some real-world testing.
...
+
+-   ast
+-   break
+-   chore
+-   cli
+-   docs
+-   feat
+-   fix
+-   misc
+-   tests
```
Comment (evilebottnawi) [inline CONTRIBUTING.md]: We use conventional commits so our type is other https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional

Comment (anikethsaha) [reply CONTRIBUTING.md]: done 👍

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Comment by hiroppy** [comment]
I think `Commit Message Format` section is unnecessary because `Commit message` has already a description link and we have commsitlint.

**Comment by anikethsaha** [comment]
> I think Commit Message Format section is unnecessary because Commit message has already a description link and we have commitlint.

agreed 👍

**Pull request closed without merging**